geode-for-redis/src/main/java/org/apache/geode/redis/internal/statistics/RedisStats.java

```diff
@@ -222,28 +221,25 @@ private void stopRollingAverageUpdater() {
   }
 
   private void doRollingAverageUpdates() {
-    updateNetworkKilobytesReadLastSecond(rollingAverageTick);
-    updateOpsPerformedOverLastSecond(rollingAverageTick);
+    networkKiloBytesReadOverLastSecond = networkKiloBytesReadRollingAverageStat
+        .calculate(getTotalNetworkBytesRead(), rollingAverageTick) / 1024.0;
+    opsPerformedOverLastSecond =
+        opsPerformedRollingAverageStat.calculate(getCommandsProcessed(), rollingAverageTick);
     rollingAverageTick++;
-    if (rollingAverageTick >= rollingAverageSamplesPerSecond) {
+    if (rollingAverageTick >= ROLLING_AVERAGE_SAMPLES_PER_SECOND) {
       rollingAverageTick = 0;
     }
   }
 
-  private void updateNetworkKilobytesReadLastSecond(int tickNumber) {
-    final long totalNetworkBytesRead = getTotalNetworkBytesRead();
-    long deltaNetworkBytesRead = totalNetworkBytesRead - totalNetworkBytesReadLastTick;
-    networkBytesReadOverLastNSamples[tickNumber] = deltaNetworkBytesRead;
-    networkKiloBytesReadOverLastSecond =
-        Arrays.stream(networkBytesReadOverLastNSamples).sum() / 1024.0;
-    totalNetworkBytesReadLastTick = totalNetworkBytesRead;
-  }
+  private static class RollingUpgradeStat {
```

Comment:
should this be named "RollingAverageStat"?

Reply:
Yes, I was apparently on auto-pilot when I typed that, thanks for catching it